### ISSUE TITLE
Syntax highlighter does not properly support Ruby percent strings

### ISSUE THREAD

**Issue by HertzDevil** [notepad-plus-plus/notepad-plus-plus#4964]
### Description of the Issue
Certain Ruby string literals beginning with `%` do not apply or suppress syntax highlighting inside the string contents properly.

### Steps to Reproduce the Issue
On any file with Ruby syntax highlighting enabled, enter the following:

```ruby
%{abc def #{1} ghi}
%i{abc def #{1} ghi}
%q{abc def #{1} ghi}
%r{abc def #{1} ghi}
%s{abc def #{1} ghi}
%w{abc def #{1} ghi}
%x{abc def #{1} ghi}
%I{abc def #{1} ghi}
%Q{abc def #{1} ghi}
%W{abc def #{1} ghi}
```

### Expected Behavior

* `%`, `%q`, and `%Q` should be highlighted as strings, `%r` as regular expressions, `%s` as symbols, and `%x` as backticks.
* String interpolation should be disabled in only `%i`, `%q`, `%s`, and `%w`.

### Actual Behavior
* These literals are not supported at all: `%i`, `%s`, `%I`. String interpolation is treated as comments, and `def` is treated as a keyword.
* `%w` does not suppress string interpolation.
* `%` and `%Q` do not display themselves like interpolated strings. `%r` and `%x` do not apply the correct colors.

### Debug Information
Notepad++ v7.5.8   (32-bit)
Build time : Jul 23 2018 - 02:03:53
Path : C:\Program Files (x86)\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS : Windows 10 (64-bit)
Plugins : MarkdownViewerPlusPlus.dll NppEditorConfig.dll NppFTP.dll NPPJSONViewer.dll PluginManager.dll zoomdisabler.dll mimeTools.dll NppConverter.dll NppExport.dll 

### Reference

* [Section on percent strings from Ruby 2.1.0](https://ruby-doc.org/core-2.1.0/doc/syntax/literals_rdoc.html#label-Percent+Strings)
  * `%R`, `%S`, and `%X` do not actually exist.
  * `%r` and `%x` do accept string interpolations, only `%s` doesn't,

**Comment by ArkadiuszMichalski** [comment]
https://sourceforge.net/p/scintilla/bugs/2182/

**Comment by zufuliu** [comment]
Lexilla issue: https://github.com/ScintillaOrg/lexilla/issues/124

**Comment by rdipardo** [comment]
@notepad-plus-plus, this was fixed in Lexilla 5.2.3 and can be closed: ScintillaOrg/lexilla@397dfb1ff667f39177ba047a990ad54b6f0de90b

**Comment by chcg** [comment]
Fix available since N++ release v8.5.1 .
